document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java

```diff
@@ -43,11 +49,48 @@ public DocumentInfo getDocumentInfo() {
                 // company accounts should work for abridged, resulting in this abridged specific code
                 // qualifying for removal
                 .filter(this::isAbridged)
-                .map(accountsLinks -> new DocumentInfo())
+                .map(accountsLinks -> getAbridgedAccountData(resourceId))
```

Comment:
How is this scalable to support other account types?

Reply:
It's pretty scalable as there is only two directions it can go. Either abridged or company-accounts. That covers all account types. The company-accounts direction will exist in the `orElse`

Reply:
What about supporting small full, micro, dormant, etc?

Reply:
That can be done in 1 method as their data is all coming from 1 api. The only reason for it splitting up is because the `abridged` data comes from a separate api. Additionally, in future, if there is more than 1 direction (which after the abridged migration - we haven't recognised one), you can add that here using multiple filters and maps.